### ISSUE TITLE
wingetcreate doesn't support multiple nodes from single installer

### ISSUE THREAD

**Issue by dpprdan** [microsoft/winget-create#166]
# Brief description of your issue

wingetcreate doesn't support multiple nodes from a single installer.

The [PostgreSQL.pgAdmin v5.6 installer manifest](https://github.com/microsoft/winget-pkgs/blob/master/manifests/p/PostgreSQL/pgAdmin/5.6/PostgreSQL.pgAdmin.installer.yaml) for example contains two installer nodes with the same URL, one for user and machine scope each, as [advised here](https://github.com/microsoft/winget-pkgs/pull/18505#issuecomment-870672541).

When trying to update this manifest to the latest version, I get the error message "The number of new installer packages must match the number of existing installer packages.". 

# Steps to reproduce

Run the following to update the PostgreSQL.pgAdmin manifest to the latest version. 

```pwsh
wingetcreate update PostgreSQL.pgAdmin --version 5.7 --urls https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v5.7/windows/pgadmin4-5.7-x64.exe
```

# Expected behavior

Update the manifest to the latest version as specified. 

# Actual behavior

Error message: "The number of new installer packages must match the number of existing installer packages." 

# Environment

```none
Windows Manifest Creator v0.3.0.6
```

**Comment by ryfu-msft** [comment]
Hey @dpprdan,

Thanks for filing this issue. We don't currently support updating multiple nodes from a single installer as it is not always clear which nodes should be updated with the new installer. However, we just published our newest release of Winget-Create [v4.0.3.0](https://github.com/microsoft/winget-create/releases/tag/v0.4.0.3-preview) which has a new interactive feature for the update command. You can now manually input the installer url you want to update a given installer node with. Hopefully that resolves your issue.

Thanks!

**Comment by ldennington** [comment]
FWIW I am also running into this issue with a CI scenario. Matching on Scope seems the most obvious choice for making this work, but unfortunately I haven't found a way to specify that when building my Inno installers.

**Comment by jedieaston** [comment]
It seems like `wingetcreate` could have a special case for this by checking if multiple of the same URL exist in the current manifest, then at least one of the installer URLs passed must be able to replace all of those.

**Comment by ldennington** [comment]
Note that our URLs are slightly different (one uses the name `gcmcoreuser` and the other `gcmcore`). For example:

https://github.com/microsoft/Git-Credential-Manager-Core/releases/download/v2.0.567/gcmcore-win-x86-2.0.567.18224.exe
https://github.com/microsoft/Git-Credential-Manager-Core/releases/download/v2.0.567/gcmcoreuser-win-x86-2.0.567.18224.exe

**Comment by jedieaston** [comment]
In the exe manifest (at least for Inno Setup, I don't know if everyone cleanly sets this), this key stuck out to me:

```
<security>
    <requestedPrivileges>
        <requestedExecutionLevel level="asInvoker" uiAccess="false" />
    </requestedPrivileges>
</security>
```
It determines whether the process is to be elevated. Couldn't we use this to determine scope?

**Comment by mdanish-kh** [comment]
Resolved with the introduction of architecture ([#206](https://github.com/microsoft/winget-create/pull/206)) and scope (https://github.com/microsoft/winget-create/pull/367) override syntax 

You would be able to use a command like

`wingetcreate update --urls "https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v5.7/windows/pgadmin4-5.7-x64.exe|x64|machine" "https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v5.7/windows/pgadmin4-5.7-x64.exe|x64|user"`

once https://github.com/microsoft/winget-create/pull/367 makes its way into a release of winget-create

cc @denelon

**Comment by cderv** [comment]
@dpprdan was that the missing piece to automate the release process ? Do you plan to test this new feature for next release ?

**Comment by dpprdan** [comment]
@cderv you mean missing with respect to https://github.com/quarto-dev/quarto-release-bundles/issues/1? I don't know to be honest.

**Comment by cderv** [comment]
yeah that it what I meant. I'll look it into before 1.4 release once new winget is released.

**Comment by mdanish-kh** [comment]
cc @denelon Resolved in WinGet-Create 1.2.6.0
